File: P08R01_Microservicios_AEHM/BACKEND/EPP.MOVIMIENTOS.API/EPP_Movimientos_API/Models/Movimiento.cs
using System.ComponentModel.DataAnnotations;

namespace EPP_Movimientos_API.Models
{
    public class Movimiento
    {
        [Key]
        public int MovimientoId { get; set; }
        [Required]
        public string? TipoMovimiento { get; set; }
        public int ProductoId { get; set; }
        [Required]
        public string? ProductoNombre { get; set; }
        public int? EmpleadoId { get; set; }
        public int Cantidad { get; set; }
        public string? Motivo { get; set; }
        public DateTime FechaMovimiento { get; set; }
        public string? RegistradoPor { get; set; }
        public Empleado? Empleado { get; set; }
    }
}

File: P08R01_Microservicios_AEHM/FRONTEND/epp-frontend/src/app/models/movimiento.ts
export class Movimiento {}

export interface Empleado{
    empleadoID?: number;
    numEmpleado: string;  //ENTRADA SALIDA
    nombre: string;
    apellidos: string;
    departamento?: string;
    puesto: string;
    activo?: boolean;
    fechaAlta?: string;
}

export interface Movimiento{
    movimientoId?: number;
    tipoMovimiento: string;  //ENTRADA SALIDA
    productoId: number;
    productoNombre: string;
    empleadoId?: number;
    cantidad: number;
    motivo?: string;
    fechaMovimiento?: string;
    registradoPor?: string;
    empleado?: Empleado;
}
